### PR TITLE
added multiple host network aliases for the same interface

### LINKED ISSUE #10106: support multiple host_network names for the same interface
### Nomad version
nomad v1.0.4

### Issue
Having issue with multi-region infrastructure setup. One region is in AWS VPC and another is in Hetzner + [Nebula](https://github.com/slackhq/nebula) for private networks setup. Host networks configuration in Hetzner is:
```
{
  "host_network": {
    "private": {
      "cidr": "<nebula-ip-address>/32"
    },
    "public": {
      "cidr": "<hetzner-ip-address>/32"
    }
  }
}
```

To be able to use the same job for both regions (cause there's no `host_network` variable interpolation) had to use the same interface for both public and private agent networks configuration in AWS for compatibility
```
{
  "host_network": {
    "private": {
      "cidr": "<awsvpc-ip-address>/32"
    },
    "public": {
      "cidr": "<awsvpc-ip-address>/32"
    }
  }
}
```
But nomad doesn't create allocations in AWS a job, which uses both public and private interfaces, cause nomad agent is registred with `private` network only. Is there any option to either add `host_network` interpolation support or use multiple network names for the same interface?

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
FIxes #10106

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -2439,7 +2441,7 @@ const (
 
 type NodeNetworkAddress struct {
 	Family        NodeNetworkAF
-	Alias         string
+	Aliases       []string
```
Comment (tgross) [inline nomad/structs/structs.go:2444]: Unfortunately this approach breaks backwards compatibility for nodes that are already saved in raft. Given that we can already store multiple `NodeNetworkAddress` for a node, we can solve this problem by writing one for each alias from the node fingerprint.

**Review by tgross — CHANGES_REQUESTED** [review]

**Comment by AndrewChubatiuk** [comment]
@tgross I've updated it

**Comment by tgross** [comment]
LGTM. I've pushed a small style change to the test; you copied what was nearby which was totally the right thing for you to do but we've been using `require` in new tests over the last year or two and have been happy with it.

**Review by tgross — APPROVED** [review]

**Comment by AndrewChubatiuk** [comment]
@tgross 
okay, thank you

**Pull request merged**